packages/kbn-es/src/utils/docker.ts

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import chalk from 'chalk';
+import execa from 'execa';
+import fs from 'fs';
+import Fsp from 'fs/promises';
+import { resolve } from 'path';
+
+import { ToolingLog } from '@kbn/tooling-log';
+import { kibanaPackageJson as pkg } from '@kbn/repo-info';
+import { getDataPath } from '@kbn/utils';
+
+import { createCliError } from '../errors';
+import { EsClusterExecOptions } from '../cluster_exec_options';
+
+interface BaseOptions {
+  tag?: string;
+  image?: string;
+}
+
+export interface DockerOptions extends EsClusterExecOptions, BaseOptions {
+  dockerCmd?: string;
+}
+
+export interface ServerlessOptions extends EsClusterExecOptions, BaseOptions {
+  clean?: boolean;
+}
+
+interface RunServerlessEsNodeArgs {
+  params: string[];
+  name: string;
+  image: string;
+}
+
+const DOCKER_REGISTRY = 'docker.elastic.co';
+
+const DOCKER_BASE_CMD = [
+  'run',
+
+  '--rm',
+
+  '--detach',
+
+  '--net',
+  'elastic',
+
+  '--name',
+  'es01',
+
+  '-p',
+  '9200:9200',
+
+  '-p',
+  '9300:9300',
+];
+
+export const DOCKER_REPO = `${DOCKER_REGISTRY}/elasticsearch/elasticsearch`;
+export const DOCKER_TAG = `${pkg.version}-SNAPSHOT`;
+export const DOCKER_IMG = `${DOCKER_REPO}:${DOCKER_TAG}`;
+export const DEFAULT_DOCKER_CMD = `${DOCKER_BASE_CMD.join(' ')} ${DOCKER_IMG}`;
+
+export const SERVERLESS_REPO = `${DOCKER_REGISTRY}/elasticsearch-ci/elasticsearch-serverless`;
+export const SERVERLESS_TAG = 'latest';
```

Comment:
I'm thinking we'll want to eventually use the tag corresponding to the latest [elasticsearch verified snapshot](https://buildkite.com/elastic/kibana-elasticsearch-snapshot-verify) eventually or we may run into surprise breaks.  

Don't think it needs to be addressed now but making a note.